include/libtorrent/extensions.hpp

```diff
@@ -426,7 +425,7 @@ namespace libtorrent
 		virtual bool on_allowed_fast(int /*index*/) { return false; }
 		virtual bool on_request(peer_request const&) { return false; }
 		virtual bool on_piece(peer_request const& /*piece*/
-			, disk_buffer_holder& /*data*/) { return false; }
+			, char* /*buf*/) { return false; }
```

Comment:
oh, I see. you changed that. what's the ownership now? (I realize that I should have updated this in my previous patch to pass this by-value too)